### DISCUSSION TITLE
[A/L] (2021/22) Funzioni - Esercizio 11

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#251] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152315437-69649396-790c-415c-8e3f-4f96b46cb95a.png)

**Comment by LuginiAndrea** [comment]
Posso dire che g è suriettiva.
Consideriamo il fatto che Z è completamente coperto da X per definizione, di conseguenza, poichè $\forall x \in X \rightarrow y \in Y$, possiamo concludere che $\forall z \in Z \, \exists y \in Y$, quindi $g$ è suriettiva.
Dimostriamo ora, tramite esempio, che non possiamo implicare nulla di $f$ e non possiamo implicare l'iniettività di $g$.
Definiamo gli insiemi 
$$X = 2, 3, 4, 5, 7$$
$$Y = 0 \leq q \leq 5, \forall q \in \mathbf{Q}$$
$$Z = 4,9,16,25$$ 

Definiamo ora 
$$f\left(x\right) = x \textrm{ se } x \leq 5, 5 \textrm{ se } x > 5$$
Possiamo vedere come l'immagine di $X$ sià $2, 3, 4, 5$, con $f\left(5\right) = f\left(7\right)$
$f$ non è quindi iniettiva, e non è neanchè suriettiva: in fatti basta considerare $\frac{2}{3}$,  che è incluso in $Y$ ma non è compreso in $f\left(X\right)$.
Definiamo ora 
$$g\left(y\right) = \textrm{ quadrato perfetto più vicino a } q^2$$
Alcuni esempi:
$$f\left(1\right) = 1, f\left(2.8\right) = 9, f\left(3\right) = 9$$
Possiamo vedere subito come $g$ non sia iniettiva.

**Comment by Elia-Belli** [comment]
Possiamo aggiungere qualcosa su f:

Abbiamo detto che **g deve essere necessariamente suriettiva**, affinchè questo sia possibile il suo dominio deve avere cardinalità maggiore o uguale al codominio: trattandosi di una composta è necessario dire che #Im(f) $\subseteq Y$ $\geq$ #Z, perchè gli unici elementi su cui agisce g nella composta sono quelli dell'immagine di X via f.

Quindi è **necessario che #Img(f) $\geq$ #Z**.
